share/functions/funced.fish

```diff
@@ -122,6 +130,26 @@ function funced --description 'Edit function definition'
                     continue
                 end
                 echo (_ "Cancelled function editing")
+            else if test -n "$writepath"
+                if set -q _flag_save
+                    # try to write the file back
+                    # cp preserves existing permissions, though it might overwrite the owner
+                    if not cp $tmpname "$writepath" 2>&1
+                        echo (_ "Saving to original location failed; saving to user configuration instead.")
+                        set writepath $__fish_config_dir/functions/(basename "$writepath")
+                        if not cp $tmpname "$writepath"
+                            echo (_ "Saving to user configuration failed. Changes will be lost when fish is closed.")
+                        else
+                            # read it back again - this ensures that the output of `functions --details` is correct
+                            source "$writepath"
+                        end
+                    else
+                        # read it back again - this ensures that the output of `functions --details` is correct
+                        source "$writepath"
+                    end
+                else
+                    echo (_ "Warning: file containing function was edited but not saved. Changes will be lost when fish is closed.")
```

Comment:
I found this message to be a little scary, and it doesn't tell the user what to do. I wish we could suggest `funcsave`, but that reintroduces the problem we are trying to solve. I also wish we could suggest `funced --save` but that doesn't work because the original file is lost.

What would you think about prompting the user to save the function if `--save` is not passed?

    > funced fish_prompt
    ...
    Save function to /home/peter/.config/fish/functions/fish_prompt? [Y/n]  Y
    Function saved to /home/peter/.config/fish/functions/fish_prompt

Reply:
Yes, that sounds great! I think that's the missing link. I'm tempted to add that for all invocations, not just those that edit a file - what do you think?

Reply:
Yeah I agree!

Reply:
I've done that, but I'm not sure about it - breaks the otherwise simple workflow of `funced`. Anyway perhaps we can merge it and see what people think.